internal/controller/pool_controller.go

```diff
@@ -294,8 +295,21 @@ func (r *PoolReconciler) reconcileDelete(ctx context.Context, garmClient garmCli
 
 	if controllerutil.ContainsFinalizer(pool, key.PoolFinalizerName) && pool.Spec.MinIdleRunners != 0 {
 		pool.Spec.MinIdleRunners = 0
-		err := r.Update(ctx, pool)
+		pool.Spec.Enabled = false
+
+		image, err := r.getImage(ctx, pool)
 		if err != nil {
+			log.Error(err, "error getting image")
+			return r.handleUpdateError(ctx, pool, err)
+		}
+
+		_, err = updatePool(garmClient, pool, image)
```

Comment:
this line is esential and new.

in the past, we've just updated the pool-cr and then the pool got scaled down.
With this implementation we immediately scaling the pool down to zero and disable the pool as well.

And after that we try to delete all the corresponding runners.